components/builder-sessionsrv/src/migrations/2018-02-15-232946_personal_access_tokens/up.sql

```diff
@@ -0,0 +1,33 @@
+CREATE SEQUENCE IF NOT EXISTS account_tokens_id_seq;
+
+CREATE TABLE IF NOT EXISTS account_tokens (
+  id bigint PRIMARY KEY DEFAULT next_id_v1('account_tokens_id_seq'),
+  account_id bigint,
+  token text UNIQUE,
+  revoked boolean DEFAULT FALSE,
```

Comment:
Why would we keep revoked tokens in the db instead of removing the row?

Reply:
Yeah maybe I'm missing something here as well, but I can't think why we'd want to keep revoked tokens in the db.

Reply:
This is a really really good question. The design is predicated currently on the notion that the universe of revoked tokens is smaller than the universe of actual tokens.  We want token validation to not require hitting the disk (or even network), so the notion is that we have a cache of revoked tokens to check against. However, an equally valid case can be made that over time the # of revoked tokens will grow to be larger than the set of valid tokens. In either case we're going to need to cache some set of tokens - either valid or revoked.  So the net of it is - we can go either way.  I'm happy to switch it over if folks think it will be cleaner/simpler.